src/routes/delegates/tests/routes.rs

```diff
@@ -0,0 +1,263 @@
+    mock_http_client
+        .expect_get()
+        .times(1)
+        .with(eq(chain_request))
+        .return_once(move |_| {
+            Ok(Response {
+                status_code: 200,
+                body: String::from(crate::tests::json::CHAIN_INFO_RINKEBY),
+            })
+        });
+    // Mock delegates transaction service
+    let mut delegates_request = Request::new(format!(
+        "https://safe-transaction.rinkeby.staging.gnosisdev.com/api/v1/delegates/?safe={}&delegate=&delegator=&label=",
+        &safe_address));
+    delegates_request.timeout(Duration::from_millis(default_request_timeout()));
+    mock_http_client
+        .expect_get()
+        .times(1)
+        .with(eq(delegates_request))
+        .returning(move |_| {
+            Ok(Response {
+                body: String::from(super::BACKEND_LIST_DELEGATES_OF_SAFE),
+                status_code: 200,
+            })
+        });
+    let client = Client::tracked(
+        setup_rocket(
+            mock_http_client,
+            routes![super::super::routes::get_delegates],
+        )
+        .await,
+    )
+    .await
+    .expect("valid rocket instance");
+    let expected =
+        serde_json::from_str::<Page<Delegate>>(super::EXPECTED_LIST_DELEGATES_OF_SAFE).unwrap();
+    // Create request get delegates
+    let request = client
+        .get(format!("/v1/chains/{}/delegates?safe={}", 4, &safe_address))
+        .header(Header::new("Host", "test.gnosis.io/api"))
+        .header(ContentType::JSON);
+    let response = request.dispatch().await;
+    let actual_status = response.status();
+    let actual_json_body = response.into_string().await.unwrap();
+    let actual = serde_json::from_str::<Page<Delegate>>(&actual_json_body).unwrap();
+    assert_eq!(actual_status, Status::Ok);
+    assert_eq!(actual, expected);
+}
+
+/// Test add delegate to safe
+/// Check the CGW requests to the config-service and transaction-service
+/// Check the status response
+#[rocket::async_test]
+async fn add_delegate() {
+    // Mock config service
+    let mut chain_request = Request::new(config_uri!("/v1/chains/{}/", 4));
+    chain_request.timeout(Duration::from_millis(chain_info_request_timeout()));
+
+    let mut mock_http_client = MockHttpClient::new();
+    mock_http_client
+        .expect_get()
+        .times(1)
+        .with(eq(chain_request))
+        .return_once(move |_| {
+            Ok(Response {
+                status_code: 200,
+                body: String::from(crate::tests::json::CHAIN_INFO_RINKEBY),
+            })
+        });
+    // Mock transaction service delegates
+    let mut delegates_request = Request::new(format!(
+        "https://safe-transaction.rinkeby.staging.gnosisdev.com/api/v1/delegates/"
+    ));
+    let mut body_string = String::from(super::BACKEND_CREATE_DELEGATE);
+    body_string = body_string.replace('\n', "");
+    body_string = body_string.replace(' ', "");
+    delegates_request.body(Some(body_string));
```

Comment:
As a reference for the future, we need to come up with a more generic way to compare json on this level.

Reply:
I agree It's not an elegant way to compare.